### PR TITLE
Fix: memory leak on volume allocation

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR fixes a memory leak issue observed when attaching more than 10 volumes on a stopped VM and try to start it.

Could be reproduced by:
- Create a VM without starting it
- Create 14 data disks and attach them to the VM
- Start VM

Screenshot 2023-01-24 at 11 18 44

````
{
  "accountid": "cdd3f51c-9862-11ed-8965-1e00dc0000f6",
  "cmd": "org.apache.cloudstack.api.command.admin.vm.StartVMCmdByAdmin",
  "completed": "2023-01-24T15:19:50+0000",
  "created": "2023-01-24T15:19:47+0000",
  "jobid": "da7e88d1-a8c4-4d44-91b3-21cef885769f",
  "jobinstanceid": "d4fed850-c112-489a-84dd-8bb875ba0e10",
  "jobinstancetype": "VirtualMachine",
  "jobprocstatus": 0,
  "jobresult": {
    "errorcode": 530,
    "errortext": "Java heap space"
  },
  "jobresultcode": 530,
  "jobresulttype": "object",
  "jobstatus": 2,
  "userid": "cdd4db33-9862-11ed-8965-1e00dc0000f6"
}
🙈 Error: async API failed for job da7e88d1-a8c4-4d44-91b3-21cef885769f
````

This was caused by extra logging on the deployment planner. An example on a VM deployment with 1 datadisk:

Before the fix:
````
2023-01-25 10:24:30,203 DEBUG [c.c.d.DeploymentPlanningManagerImpl] (API-Job-Executor-31:ctx-7c14f8a1 job-228 ctx-bd96ad97) (logid:b91e0e7d) Returning Deployment Destination: Dest[Zone(Id)-Pod(Id)-Cluster(Id)-Host(Id)-Storage(Volume(Id|Type-->Pool(Id))] : Dest[Zone(1)-Pod(1)-Cluster(1)-Host(2)-Storage(Volume(82|ROOT-->Pool(2)Volume(82|ROOT-->Pool(2), Volume(82|ROOT-->Pool(2)Volume(82|ROOT-->Pool(2), Volume(83|DATADISK-->Pool(1))]
````

After the fix:
````
2023-01-25 11:18:09,612 DEBUG [c.c.d.DeploymentPlanningManagerImpl] (API-Job-Executor-1:ctx-9f7540cf job-234 ctx-9db31cfd) (logid:50df4eb2) Returning Deployment Destination: Dest[Zone(Id)-Pod(Id)-Cluster(Id)-Host(Id)-Storage(Volume(Id|Type-->Pool(Id))] : Dest[Zone(1)-Pod(1)-Cluster(1)-Host(2)-Storage(Volume(84|ROOT-->Pool(2)Volume(85|DATADISK-->Pool(1))]
````

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Review by weizhouapache — APPROVED** [review]
code lgtm

not tested yet

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by slavkap — APPROVED** [review]
LGTM tested. Reproduced it on 4.16/main branch

**Comment by nvazquez** [comment]
@weizhouapache @slavkap I have changed the base branch to 4.17 and your reviews got reset, can you please re-review?

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Review by slavkap — APPROVED** [review]
I didn't run the Marvin test but the code LGTM

File: test/integration/smoke/test_vm_life_cycle.py [hunk 1]
```diff
@@ -873,6 +873,70 @@ def test_11_destroy_vm_and_volumes(self):
...
+        )
+        self.assertEqual(
+            list_vm_response[0].state,
+            "Running",
+            "Check virtual machine is in running state"
+        )
+
+        # Destroy VM and volumes
+        self.debug("Destroy VM - ID: %s" % vm.id)
+        vm.delete(self.apiclient, volumeIds=volume_ids)
```
Comment (DaanHoogland) [inline test/integration/smoke/test_vm_life_cycle.py:938]: if any of the asserts trigger this is not executed and we leave garbage. Rather use `cleanup.append()` on all the volumes and the VM.

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
Can be merged if smoketests pass. @nvazquez can you check if this new test be added to Github Actions?

**Review by stephankruggg — APPROVED** [review]
CLGTM

**Comment by DaanHoogland** [comment]
@nvazquez cc @rohityadavcloud the new test failed in travis, can you have a look?

**Comment by nvazquez** [comment]
@DaanHoogland I've checked the test fails on Travis since the datadisks to attach to the VM in the new test exceeds the maximum number of datadisks, working on a fix.

````
Printing marvin logs /tmp//MarvinLogs/test_vm_life_cycle_UYSWQ0/failed_plus_exceptions.txt :
2023-01-27 03:38:49,262 - CRITICAL - EXCEPTION: test_12_start_vm_multiple_volumes_allocated: ['Traceback (most recent call last):\n', '  File "/opt/pyenv/versions/3.6/lib/python3.6/unittest/case.py", line 59, in testPartExecutor\n    yield\n', '  File "/opt/pyenv/versions/3.6/lib/python3.6/unittest/case.py", line 605, in run\n    testMethod()\n', '  File "/home/travis/build/apache/cloudstack/test/integration/smoke/test_vm_life_cycle.py", line 911, in test_12_start_vm_multiple_volumes_allocated\n    VirtualMachine.attach_volume(vm, self.apiclient, volume)\n', '  File "/home/travis/.local/lib/python3.6/site-packages/marvin/lib/base.py", line 895, in attach_volume\n    return apiclient.attachVolume(cmd)\n', '  File "/home/travis/.local/lib/python3.6/site-packages/marvin/cloudstackAPI/cloudstackAPIClient.py", line 2741, in attachVolume\n    response = self.connection.marvinRequest(command, response_type=response, method=method)\n', '  File "/home/travis/.local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 381, in marvinRequest\n    raise e\n', '  File "/home/travis/.local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 376, in marvinRequest\n    raise self.__lastError\n', '  File "/home/travis/.local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 105, in __poll\n    % async_response)\n', "Exception: Job failed: {accountid : '9671e93c-9df2-11ed-bf74-42010a1e0014', userid : '967213df-9df2-11ed-bf74-42010a1e0014', cmd : 'org.apache.cloudstack.api.command.admin.volume.AttachVolumeCmdByAdmin', jobstatus : 2, jobprocstatus : 0, jobresultcode : 530, jobresulttype : 'object', jobresult : {errorcode : 431, errortext : 'The specified VM already has the maximum number of data disks (6) attached. Please specify another VM.'}, jobinstancetype : 'Volume', jobinstanceid : '978938a2-693a-4f7e-bcfd-3f45045d2405', created : '2023-01-27T03:38:48+0000', completed : '2023-01-27T03:38:48+0000', jobid : '0dee5876-034e-4384-8d48-22de3eb7c741'}\n"]
````

**Comment by nvazquez** [comment]
@blueorangutan package

File: test/integration/smoke/test_vm_life_cycle.py [hunk 2]
```diff
@@ -895,11 +895,39 @@ def test_12_start_vm_multiple_volumes_allocated(self):
...
+            self.apiclient,
+            zoneid=self.zone.id,
+            type='Routing',
+            hypervisor=self.hypervisor,
+            state='Up')
+
+        if self.hypervisor.lower() in ["simulator"]:
+            hypervisor_version = "default"
+        else:
+            res = self.dbclient.execute("select hypervisor_version from host where uuid = '%s'" % hosts[0].id)
```
Comment (slavkap) [inline test/integration/smoke/test_vm_life_cycle.py:908]: There is no need for this query. The hypervisor version is a response parameter of list hosts

File: test/integration/smoke/test_vm_life_cycle.py (hunk 1 above)
Comment (nvazquez) [reply test/integration/smoke/test_vm_life_cycle.py]: Done, thanks

File: test/integration/smoke/test_vm_life_cycle.py (hunk 2 above)
Comment (nvazquez) [reply test/integration/smoke/test_vm_life_cycle.py]: Done, thanks

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test matrix

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test matrix

**Pull request merged**